server/protocols/protocols-imap4/src/main/java/org/apache/james/imapserver/netty/IMAPServer.java

```diff
@@ -209,6 +252,7 @@ protected String getDefaultJMXName() {
     protected ChannelUpstreamHandler createCoreHandler() {
         ImapChannelUpstreamHandler coreHandler;
         Encryption secure = getEncryption();
+        boolean plainAuthDisallowed = evaluatePlainAuthDisallowed();
         if (secure != null && secure.isStartTLS()) {
            coreHandler = new ImapChannelUpstreamHandler(hello, processor, encoder, compress, plainAuthDisallowed, secure, imapMetrics);
```

Comment:
```suggestion
           coreHandler = new ImapChannelUpstreamHandler(hello, processor, encoder, compress, plainAuthDisallowed, secure, imapMetrics);
```

IMO we should pass the full authentication configuration rather than individual parameters.